### PR TITLE
resource/aws_route53_resolver_firewall_rule_group_association: Add new resource

### PR DESCRIPTION
**Pull request by gazoakley**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #18520

Output from acceptance testing:

```
$ gaz@Gaz-MacBook-Pro terraform-provider-aws % make testacc TESTARGS='-run=TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_'                  
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_ -timeout 180m
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_basic
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_basic
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_name
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_name
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_mutationProtection
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_mutationProtection
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_priority
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_priority
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_disappears
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_disappears
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_tags
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_tags
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_basic
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_disappears
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_tags
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_name
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_priority
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_mutationProtection
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_disappears (206.91s)
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_basic (218.45s)
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_mutationProtection (244.96s)
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_tags (282.00s)
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_priority (304.53s)
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_name (395.29s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	399.240s
```

### CODE REVIEW

File: aws/resource_aws_route53_resolver_firewall_rule_group_association.go [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+	if ruleGroupAssociation == nil {
+		log.Printf("[WARN] Route 53 Resolver DNS Firewall rule group association (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	arn := aws.StringValue(ruleGroupAssociation.Arn)
+	d.Set("arn", arn)
+	d.Set("id", ruleGroupAssociation.Id)
```
Comment (ewbankkit) [inline aws/resource_aws_route53_resolver_firewall_rule_group_association.go:135]: Remove `id` attribute.

File: aws/resource_aws_route53_resolver_firewall_rule_group_association_test.go [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+
+func testSweepRoute53ResolverFirewallRuleGroupAssociations(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).route53resolverconn
+	var sweeperErrs *multierror.Error
+
+	err = conn.ListFirewallRuleGroupAssociationsPages(&route53resolver.ListFirewallRuleGroupAssociationsInput{}, func(page *route53resolver.ListFirewallRuleGroupAssociationsOutput, isLast bool) bool {
```
Comment (ewbankkit) [inline aws/resource_aws_route53_resolver_firewall_rule_group_association_test.go:32]: Change `isLast` to `lastPAge` (new semgrep rule).

**Comment by ewbankkit** [comment]
@gazoakley Thanks for the contribution 👏.
I've left a couple of review items and will give it a more thorough look over once you resolve the merge conflicts.
Could you please add a `CHANGELOG` entry file as documented in the [Contributing Guide](https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/pullrequest-submission-and-lifecycle.md#changelog-process)?
Thanks.

**Comment by gazoakley** [comment]
Hi @ewbankkit - thanks for the review 👍 . I've fixed the merge conflicts and addressed the review comments

**Comment by ewbankkit** [comment]
@gazoakley Everything seems on the right track here :clap:.
If you have time, could you please add `default_tags` support (#7926) to the new resources (see #19084 for documentation updates on how to do this or for some examples, #18726)? If not, let us know and I can add this pre-merge.

**Review by ewbankkit — APPROVED** [review]
LGTM 🚀.

##### Commercial

```console
% make testacc TEST=./aws TESTARGS='-run=TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_ -timeout 180m
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_basic
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_basic
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_name
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_name
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_mutationProtection
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_mutationProtection
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_priority
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_priority
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_disappears
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_disappears
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_tags
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_tags
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_basic
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_disappears
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_tags
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_priority
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_name
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_mutationProtection
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_disappears (136.95s)
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_basic (150.81s)
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_mutationProtection (156.45s)
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_tags (193.68s)
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_priority (246.86s)
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_name (326.82s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	329.687s
```

##### GovCloud

```console
% make testacc TEST=./aws TESTARGS='-run=TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_ -timeout 180m
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_basic
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_basic
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_name
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_name
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_mutationProtection
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_mutationProtection
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_priority
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_priority
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_disappears
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_disappears
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_tags
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_tags
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_basic
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_disappears
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_priority
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_name
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_mutationProtection
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_basic
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_tags
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_basic (204.45s)
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_tags (224.56s)
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_disappears (172.02s)
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_mutationProtection (218.30s)
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_priority (285.65s)
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroupAssociation_name (419.11s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	422.046s
```

**Comment by ewbankkit** [comment]
@gazoakley Sorry, some sort of git mess up on my part got this one closed and I can't reopen it.
I have moved your commit into #19164 and will merge that one.

**Pull request closed without merging**